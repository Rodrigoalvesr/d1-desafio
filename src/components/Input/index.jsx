import React from 'react';
import PropType from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import InputBase from '@material-ui/core/InputBase';
import classnames from 'classnames';

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    outline: 'none',
    border: '1px solid #CCCFDE',
    padding: 6,
    borderRadius: 5,
  },
  icon: {
    display: 'flex',
    alignItems: 'center',
    marginRight: theme.spacing(1),
    marginLeft: theme.spacing(1),
    '& svg': {
      fill: theme.palette.primary.main,
    },
  },
  input: {
    fontSize: 12,
    '&::placeholder': {
      color: '#9196AB',
    },
  },
}));

function Input({ startIcon, className, ...rest }) {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      {startIcon && <div className={classes.icon}>{startIcon}</div>}
      <InputBase
        fullWidth
        className={classnames(classes.input, className)}
        {...rest}
      />
    </div>
  );
}

Input.propTypes = {
  startIcon: PropType.node,
  className: PropType.string,
};

export default Input;
