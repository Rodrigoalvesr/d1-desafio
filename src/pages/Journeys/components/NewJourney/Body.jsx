import React from 'react';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import Input from '@Components/Input';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles({
  root: {
    marginBottom: 15,
  },
  text: {
    color: '#3E4157',
    fontSize: 13,
  },
  helper: {
    color: '#9196AB',
    fontSize: 13,
  },
});

function Body({ text, setText }) {
  const classes = useStyles();

  const handleChange = React.useCallback(
    (event) => setText(event.target.value),
    [setText]
  );

  return (
    <div className={classes.root}>
      <Typography className={classes.text}>
        Dê um <strong>nome</strong> para essa Jornada
      </Typography>
      <Input value={text} onChange={handleChange} className={classes.input} />
      <Typography className={classes.helper}>
        Você poderá alterar essa informação depois.
      </Typography>
    </div>
  );
}

Body.propTypes = {
  text: PropTypes.string.isRequired,
  setText: PropTypes.func.isRequired,
};

export default Body;
