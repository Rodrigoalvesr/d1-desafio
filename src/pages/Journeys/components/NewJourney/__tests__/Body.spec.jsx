import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Body from '../Body';
import Input from '@Components/Input';

describe('Body', () => {
  it('Renders correctly', () => {
    const component = renderer
      .create(<Body text="" setText={jest.fn()} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('Should change the input text and call the mock function', () => {
    const mock = jest.fn();
    const event = {
      preventDefault: jest.fn(),
      target: { value: 'new value' },
    };
    const body = shallow(<Body text="" setText={mock} />);
    const input = body.find(Input);
    input.simulate('change', event);
    expect(mock.mock.calls.length).toEqual(1);
    expect(mock).toBeCalledWith('new value');
  });
});
