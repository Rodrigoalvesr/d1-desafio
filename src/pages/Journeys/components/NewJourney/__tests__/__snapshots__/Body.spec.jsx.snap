// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body Renders correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <p
    className="MuiTypography-root makeStyles-text-2 MuiTypography-body1"
  >
    Dê um 
    <strong>
      nome
    </strong>
     para essa Jornada
  </p>
  <div
    className="makeStyles-root-4"
  >
    <div
      className="MuiInputBase-root makeStyles-input-6 MuiInputBase-fullWidth"
      onClick={[Function]}
    >
      <input
        className="MuiInputBase-input"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
        value=""
      />
    </div>
  </div>
  <p
    className="MuiTypography-root makeStyles-helper-3 MuiTypography-body1"
  >
    Você poderá alterar essa informação depois.
  </p>
</div>
`;
